app/controllers/WhenToSendSampleController.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import connectors.DataCacheConnector
+import controllers.actions._
+import config.FrontendAppConfig
+import forms.WhenToSendSampleFormProvider
+import models.{Enumerable, Mode}
+import pages.{WhenToSendSamplePage, ReturnSamplesPage, SimilarItemCommodityCodePage}
+import navigation.Navigator
+import views.html.whenToSendSample
+import models.WhenToSendSample.{Yesprovidesample, Notsendingsample}
+
+import scala.concurrent.Future
+
+class WhenToSendSampleController @Inject()(
+                                        appConfig: FrontendAppConfig,
+                                        override val messagesApi: MessagesApi,
+                                        dataCacheConnector: DataCacheConnector,
+                                        navigator: Navigator,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: WhenToSendSampleFormProvider
+                                      ) extends FrontendController with I18nSupport with Enumerable.Implicits {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode) = (identify andThen getData andThen requireData) {
+    implicit request =>
+
+      val preparedForm = request.userAnswers.get(WhenToSendSamplePage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      Ok(whenToSendSample(appConfig, preparedForm, mode))
+  }
+
+  def onSubmit(mode: Mode) = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      form.bindFromRequest().fold(
+        (formWithErrors: Form[_]) =>
+          Future.successful(BadRequest(whenToSendSample(appConfig, formWithErrors, mode))),
+        (value) => {
+          val updatedAnswers = request.userAnswers.set(WhenToSendSamplePage, value)
+
+          val redirectedPage = value match {
+            case Yesprovidesample => ReturnSamplesPage
+            case Notsendingsample => SimilarItemCommodityCodePage
```

Comment:
Is the case-ing of `Notsendingsample` a Scaffold thing? Usually we'd capitalize each word like `NotSensingSample`

Reply:
yes, I used camel case when creating the page using scaffold, but scaffold generates these in the models files. As you can see here - https://github.com/hmrc/binding-tariff-trader-frontend/blob/DIT-377/app/models/WhenToSendSample.scala#L27-#L28

Reply:
Fair enough :)